### PR TITLE
Refactor image and images data source and documentation

### PR DESCRIPTION
**Pull request by Diptipowervs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by Diptipowervs** [comment]
--- PASS: TestAccIBMPIImageDataSource_basic (38.56s)
PASS

**Review by yussufsh — APPROVED** [review]
lgtm

File: ibm/service/power/data_source_ibm_pi_image.go [hunk 1]
```diff
@@ -6,65 +6,72 @@ package power
...
 			},
-			"architecture": {
-				Type:     schema.TypeString,
-				Computed: true,
+			Attr_ImageType: {
+				Computed:    true,
+				Description: "The identifier of this image type.",
+				Type:        schema.TypeString,
 			},
+			// TODO: Relabel this one "operating_system" to match catalog images
```
[the remark this answers is not in the document]
Comment (ismirlia) [reply ibm/service/power/data_source_ibm_pi_image.go:50]: Yea I know I was just frustrated when I found it.

**Review by michaelkad — APPROVED** [review]
Lgtm

**Pull request merged**